samples/sample_multi_transcode/src/sample_multi_transcode.cpp

```diff
@@ -358,36 +359,32 @@ void Launcher::DoTranscoding()
 {
     mfxStatus sts = MFX_ERR_NONE;
 
-    // get parallel sessions parameters
-    mfxU32 totalSessions = (mfxU32)m_pSessionArray.size();
-    MSDKThread* pthread = 0;
-
-    for (mfxU32 i = 0; i < totalSessions; i++)
+    // Run transcoding threads
+    std::vector<std::future<mfxU32>> thrHandles;
+    thrHandles.reserve(m_pSessionArray.size());
+    for (auto session : m_pSessionArray)
     {
-        pthread = new MSDKThread(sts, TranscodeRoutine, (void *)m_pSessionArray[i]);
-        m_HDLArray.push_back(pthread);
-    }
-
-    // Need to determine overlay threads count first
-    size_t nOverlayThreads = 0;
-    for (size_t i = 0; i < m_pSessionArray.size(); i++)
-    {
-        if (m_pSessionArray[i]->pPipeline->IsOverlayUsed())
-            nOverlayThreads++;
+        thrHandles.emplace_back(std::async(std::launch::async,
+                                TranscodeRoutine, (void *)session));
     }
 
     // Transcoding threads waiting cycle
+    size_t totalSessions = m_pSessionArray.size();
     size_t aliveSessionsCount = totalSessions;
     while (aliveSessionsCount)
     {
         aliveSessionsCount = 0;
-        for (mfxU32 i = 0; i < totalSessions; i++)
+        size_t aliveOverlaySessionsCount = 0;
+
+        for (size_t i = 0; i < totalSessions; i++)
         {
-            if (m_HDLArray[i])
-            {
-                aliveSessionsCount++;
+            if (!thrHandles[i].valid())
+                continue;
 
-                sts = m_HDLArray[i]->TimedWait(1);
+            auto waitSts = thrHandles[i].wait_for(std::chrono::milliseconds(1));
+            if (waitSts == std::future_status::ready)
+            {
+                auto sts = thrHandles[i].get();
```

Comment:
this one would be zero anyway... see return of
```
mfxU32 MFX_STDCALL TranscodingSample::TranscodeRoutine(void *pObj)
```
also, returned status of thread routine here doesn't make any sense, because we track errors by transcodingSts. I suggest to make ```TranscodeRoutine``` return ```void```

Reply:
Done